pkg/controller/operators/catalog/operator.go

```diff
@@ -2040,11 +2044,12 @@ func (o *Operator) apiresourceFromGVK(gvk schema.GroupVersionKind) (metav1.APIRe
 	}
 	for _, r := range resources.APIResources {
 		if r.Kind == gvk.Kind {
+			logger.WithField("kind", r.Kind).Infof("found GVK %s in api discovery", gvk.String())
 			return r, nil
 		}
 	}
 	logger.Info("couldn't find GVK in api discovery")
-	return metav1.APIResource{}, olmerrors.GroupVersionKindNotFoundError{gvk.Group, gvk.Version, gvk.Kind}
+	return metav1.APIResource{}, fmt.Errorf("could not find GVK %s in api discovery", gvk.String())
```

Comment:
IT looks like OLM will keep trying to reconcile the CR if the CRD is not present. What happens if the CR's CRD is not present on cluster and is not included in the installPlan. Should OLM fail the CSV?

Reply:
That's essentially more concern in https://github.com/operator-framework/operator-lifecycle-manager/pull/1708#issuecomment-670140773

How can we tell whether want to fail the CSV or wait longer for the GVK to become available

Reply:
we could set a `context.WithTimeout()` set to like 5 minutes?

Reply:
Prototyped this out in https://github.com/exdx/operator-lifecycle-manager/pull/3